### PR TITLE
update CVE-2021-26084 detector

### PR DESCRIPTION
**Pull request by C4o**
Hi @magl0 , I have completed the detector of CVE-2021-26084, please review it. Thanks!

isseue: #174 

It may take some time to build Confluence environment, cause It need license which can be free in 90 days from official website and databases like postgresql(driver installed already) or other dbs which need to install driver in confluence docker manually.

vulnerable docker-compose config
```
version: '3.1'
services:
  web:
    image: atlassian/confluence-server:7.12.4 
    ports:
      - "8080:8090"
    depends_on:
      - db
    tty: true

  db:
    image: postgres
    ports:
      - "5432:5432"
    environment:
      POSTGRES_PASSWORD: 123456 
      POSTGRES_DB: confluence 
      POSTGRES_USER: postgre
```

not-vulnerable docker-compose config
```
version: '3.1'
services:
  web:
    image: atlassian/confluence-server:7.12.5
    ports:
      - "8080:8090"
    depends_on:
      - db
    tty: true

  db:
    image: postgres
    ports:
      - "5432:5432"
    environment:
      POSTGRES_PASSWORD: 123456 
      POSTGRES_DB: confluence 
      POSTGRES_USER: postgre
```

Here is the details to run it.
When docker is up, you need to get the trial license from official website,
step1

and choose `data center`,
step2
step3
step4
step5

and you can use `db` as the alias of postgresql in installation.
step6

and then choose the simplest configuration and go on,
step7
step8

and admin's account and password and space name can be ignored cause this vulnerability is pre-auth.
step9
step10

### CODE REVIEW

File: community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetector.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "AtlassianConfluencePreAuthOGNLInjectionDetector",
+    version = "0.1",
+    description = "This detector checks for unprotected  Atlassian Confluence Pre-Auth OGNL "
+        + "Injection.",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = AtlassianConfluencePreAuthOGNLInjectionDetectorBootstrapModule.class)
+public final class AtlassianConfluencePreAuthOGNLInjectionDetector implements VulnDetector {
```
Comment (magl0) [inline community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetector.java:59]: Code style: `Ognl`

File: community/detectors/confluence_cve_2021_26084/src/test/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetectorTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ * Unit tests for {@link AtlassianConfluencePreAuthOGNLInjectionDetector}.
+ */
+@RunWith(JUnit4.class)
+public final class AtlassianConfluencePreAuthOGNLInjectionDetectorTest {
+
+  private final FakeUtcClock fakeUtcClock =
+      FakeUtcClock.create().setNow(Instant.parse("2020-01-01T00:00:00.00Z"));
+
+  @Inject
+  private  AtlassianConfluencePreAuthOGNLInjectionDetector detector;
```
Comment (magl0) [inline community/detectors/confluence_cve_2021_26084/src/test/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetectorTest.java:58]: Code style: remove the extra whitespace after `private`.

File: community/detectors/confluence_cve_2021_26084/src/test/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetectorTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Unit tests for {@link AtlassianConfluencePreAuthOGNLInjectionDetector}.
+ */
+@RunWith(JUnit4.class)
+public final class AtlassianConfluencePreAuthOGNLInjectionDetectorTest {
```
Comment (magl0) [inline community/detectors/confluence_cve_2021_26084/src/test/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetectorTest.java:52]: Code style: similar to the detector class, `Ognl`.

File: community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetectorBootstrapModule.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.detectors.confluence;
+
+import com.google.tsunami.plugin.PluginBootstrapModule;
+
+/**
+ * A {@link PluginBootstrapModule} for {@link AtlassianConfluencePreAuthOGNLInjectionDetector}
+ */
+public final class AtlassianConfluencePreAuthOGNLInjectionDetectorBootstrapModule extends
```
Comment (magl0) [inline community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetectorBootstrapModule.java:23]: Code style: similar to the detector class, `Ognl`.

File: community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetector.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final int number = 10086;
+  private static final String payload = String.format("queryString=Google\\u0027%%2b"
+      + "#{%d*%d}%%2b\\u0027Tsunami", number, number);
+  private static final String pattern = String.format("Google{%d=null}Tsunami", number * number);
```
Comment (magl0) [inline community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetector.java:65]: Code style: please use `CONSTANT_CASE` for constants (https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names).

File: community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOgnlInjectionDetector.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        + "Injection.",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = AtlassianConfluencePreAuthOgnlInjectionDetectorBootstrapModule.class)
+public final class AtlassianConfluencePreAuthOgnlInjectionDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final int NUMBER = 10086;
+  private static final String payload = String.format("queryString=Google\\u0027%%2b"
+      + "#{%d*%d}%%2b\\u0027Tsunami", NUMBER, NUMBER);
+  private static final String pattern = String.format("Google{%d=null}Tsunami", NUMBER * NUMBER);
```
Comment (magl0) [inline community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOgnlInjectionDetector.java:65]: Please updated these 2 variables to `CONSTANT_CASE` as well.

**Review by magl0 — COMMENTED** [review]
Only a minor code style comment. I'll redirect this to an internal reviewer for code reviews and kick-off the payload verification process. Feedbacks will be mirrored back to this PR.

**Comment by C4o** [comment]
Done. Thanks!

**Comment by magl0** [comment]
I've redirected this for internal code review.

**Pull request merged**